### ISSUE TITLE
ACE1 Software Paper

### ISSUE THREAD

**Issue by cortner** [ACEsuit/ACEpotentials.jl#97]
This is a META-issue collecting what are the key priorities to complete for the publication of the software paper.  I will start filling in details here and ask everybody I CC to closely follow discussions here. If there are specific issues on your mind that you'd like included here, it may be better to discuss on Slack and post below only when a discussion has settled and there is a clear TODO item. I will keep editing *this* post with updates.

0. Paper writing 
 
1. Documentation 

2. Names of user-facing functions and parameters

3. Tutorials

4. Default Heuristics 

5. Code Code and Performance

CC @tjjarvinen @WillBaldwin0 @gelzinyte @wcwitt @casv2

**Comment by cortner** [comment]
**0. Paper**
- [x] Write appendix that sketches out fast evaluation with reference to new kernel. (CO)
- [x] Fix how we present the code / pseudocode
- ~~JSON interface~~


**1. Documentation**

- ~~rewrite "intro to ACE models" once the manuscript is done and base it on that.~~ -> moved to another issue
- ~~need a page in the docs on how to prepare data to work with ACE1pack #95~~  -> moved to another issue
- [x] The ASE and LAMMMPs tutorials should be extended and self-contained.
-  ~~liberally add links into the documentation, e.g. every time we mention ACE1 or ACE1pack etc.~~

**Comment by cortner** [comment]
**2. Names of user-facing functions and parameters**

- [x] should the default key for forces be `"forces"` instead of `"force"`? @wcwitt. WCW edit: @gelzinyte and I made this change.
- [x] The smoothness prior is at the moment impossible to construct for a user, we should introduce the function `smoothness_prior`, already done for the `ACE1Model` in #92 , but should also introduce this for the ACE basis and the combined basis; done in ACE1pack, TODO move it to ACE1? Elena edit: Also rename ACE1pack's `regularizer.laplacian_regularizer` to be consistent with `ACE1Model`. 
- [x] Closely related to the previous item: should the prior be passed to the solver or should it be passed to `acefit!`?
- [x] Consider Elena's suggestions for clearer keyword argument, `maxdeg2, rcut2`; cf #99 
- [x] ACEfit Bayesian solver names and default parameters; https://github.com/ACEsuit/ACEfit.jl/issues/41
- [x] `P` vs `Gamma` for the regulariser. Probably `P` for prior, but let's please never call it a preconditioner again
- [x] `species` -> `elements` (requested by Gabor)
- [x] `N` -> `bodyorder`  or should it be `order`?  (requested by Gabor)

**Comment by cortner** [comment]
**3. Tutorials**

- [x] Fix the output of `ACE1pack.linear_errors`; cf. #98
- [x] what is the default solver and default solver parameters that we should use in the tutorials? @wcwitt 
- [x] need expanded examples (for dummies!) for ASE and LAMMPs usage, #94 ; those tutorials should be fully self-contained and need to be revised (broken links etc...)
- [x] add a tutorial on running a simple MD simulation with `Molly.jl`

**Comment by cortner** [comment]
**4. Default Heuristics**

- [x] add multi-degree and wL choices to the `acebasis` and `acemodel` interfaces - CO, done for ACE1.jl, todo for ACE1pack.jl  cf #104 
- ~~When using BLR then the solver should automagically generate a committee and attach it to the potential. Or, alternatively, this can be done in `acefit!` instead of BLR? What's the right approach? cf #106~~
- ~~BLR should be the default solver and if no solver is specified it should work reasonably well with default parameters. To be tested on a few benchmarks?~~
- [x] `maxdeg_pair = maxdeg`; i.e. hide the 2-body from the user until they dig deeper into the documentation

**Comment by cortner** [comment]
**5. Core Code and Performance**

- [x] CI needs to be fixed
- ~~consider moving to new kernels asap~~
- [x] Make ACE1 potentials compatible with `Molly.jl` - cc @tjjarvinen 
- [x] Add unit test for `acemodel` (- EG edit)

**Random stuff**
- [x] export `smoothness_prior`

**Comment by wcwitt** [comment]
Thanks for this. I'm going to add some issues here, triaging based on what's essential for the manuscript.

**Comment by wcwitt** [comment]
Thanks for this.

**Comment by cortner** [comment]
I see you edited your message. I guess it's ok to discuss here as well, but let's make sure all ACTUAL action points remain in the posts above.
